Source/src/WixSharp.UI/ManagedUI/Forms/ProgressDialog.cs

```diff
@@ -108,54 +105,37 @@ public override MessageResult ProcessMessage(InstallMessage messageType, Record
-                            if (simple)
-                            {
-                                /*
-                                messageRecord[2] unconditionally contains the string to display
-
-                                Examples:
-
-                                   messageRecord[0]    "Action 23:14:50: [1]. [2]"
-                                   messageRecord[1]    "InstallFiles"
-                                   messageRecord[2]    "Copying new files"
-                                   messageRecord[3]    "File: [1],  Directory: [9],  Size: [6]"
-
-                                   messageRecord[0]    "Action 23:15:21: [1]. [2]"
-                                   messageRecord[1]    "RegisterUser"
-                                   messageRecord[2]    "Registering user"
-                                   messageRecord[3]    "[1]"
-
-                                */
-                                if (messageRecord.FieldCount >= 3)
-                                {
-                                    message = messageRecord[2].ToString();
-                                }
-                            }
-                            else
+                            /*
+                            messageRecord[0] - is reserved for FormatString value
+                            messageRecord[2] unconditionally contains the string to display
+
+                            Examples:
+
+                               messageRecord[0]    "Action 23:14:50: [1]. [2]"
+                               messageRecord[1]    "InstallFiles"
+                               messageRecord[2]    "Copying new files"
+                               messageRecord[3]    "File: [1],  Directory: [9],  Size: [6]"
+
+                               messageRecord[0]    "Action 23:15:21: [1]. [2]"
+                               messageRecord[1]    "RegisterUser"
+                               messageRecord[2]    "Registering user"
+                               messageRecord[3]    "[1]"
+
+                            */
+                            string message = null;
+                            if (messageRecord.FieldCount >= 3)
                             {
-                                message = messageRecord.FormatString;
-                                if (message.IsNotEmpty())
-                                {
-                                    for (int i = 1; i < messageRecord.FieldCount; i++)
-                                    {
-                                        message = message.Replace("[" + i + "]", messageRecord[i].ToString());
-                                    }
-                                }
-                                else
-                                {
-                                    message = messageRecord[messageRecord.FieldCount - 1].ToString();
-                                }
+                                message = messageRecord[2].ToString();
```

Comment:
Could you solve this problem using a separate class ProgressDialogSimple that inherits from this? Might be overkill for a simple boolean, but semantically cleaner.

Reply:
But this means including two files in the template and then advising the user to chose one or another.

I prefer a simple communication channel - code.

Though there is a compromise. Can you please just comment out the `!simple` algorithm and put a comment message to the user.  Something like 
```c#
// uncomment the block below if y want to display more informative progress message
```

It will not wake up sq and yet will be even more informative for the user

Reply:
done